Sources/Moya/JSONDecoder+KeyPath.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+internal extension JSONDecoder {
+
+    /// Decode value at the keypath of the given type from the given JSON representation
+    ///
+    /// - Parameters:
+    ///   - type: The type of the value to decode.
+    ///   - data: The data to decode from.
+    ///   - keyPath: The JSON keypath
+    ///   - keyPathSeparator: Nested keypath separator
+    /// - Returns: A value of the requested type.
+    /// - Throws: An error if any value throws an error during decoding.
+    func decode<T>(_ type: T.Type,
+                   from data: Data,
+                   keyPath: String,
+                   keyPathSeparator separator: String = ".") throws -> T where T: Decodable {
+        userInfo[keyPathUserInfoKey] = keyPath.components(separatedBy: separator)
+        return try decode(KeyPathWrapper<T>.self, from: data).object
+    }
+}
+
+/// The keypath key in the `userInfo`
+private let keyPathUserInfoKey = CodingUserInfoKey(rawValue: "keyPathUserInfoKey")! //swiftlint:disable:this force_unwrapping
```

Comment:
We should have an explanation in comments for why the force-unwrap always works.